elasticsearch-vmss/azuredeploy.json

```diff
@@ -0,0 +1,504 @@
+          "count": 3
+        },
+        "properties": {
+          "ipConfigurations": [
+            {
+              "name": "ipconfig1",
+              "properties": {
+                "privateIPAllocationMethod": "Static",
+                "privateIPAddress": "[concat('10.0.0.1', copyindex())]",
+                "subnet": {
+                  "id": "[concat(resourceId('Microsoft.Network/virtualNetworks', variables('virtualNetworkName')), '/subnets/', variables('subnetName'))]"
+                },
+                "loadBalancerBackendAddressPools": [
+                    {
+                        "id": "[concat('/subscriptions/', subscription().subscriptionId,'/resourceGroups/', resourceGroup().name, '/providers/Microsoft.Network/loadBalancers/', variables('loadBalancerName'), '/backendAddressPools/', variables('backendPoolName'))]"
+                    }
+                ],
+                "networkSecurityGroup": {
+                  "id": "[resourceId('Microsoft.Network/networkSecurityGroups', variables('nsgName'))]"
+                }
+              }
+            }
+          ]
+        }
+    },
+    {
+        "apiVersion": "[variables('computeApiVersion')]",        
+        "type": "Microsoft.Compute/virtualMachines",
+        "name": "[concat(variables('masterName'), copyindex())]",
+        "location": "[resourceGroup().location]",
+        "copy": {
+          "name": "masterVmLoop",
+          "count": 3
+        },
+        "dependsOn": [
+          "[concat('Microsoft.Network/networkInterfaces/', variables('nicName'), copyindex())]",
+          "[concat('Microsoft.Compute/availabilitySets/', variables('availabilitySetName'))]",
+          "[concat('Microsoft.Storage/storageAccounts/', variables('masterStorageAccountName'))]"
+        ],
+        "properties": {
+          "availabilitySet": {
+            "id": "[resourceId('Microsoft.Compute/availabilitySets', variables('availabilitySetName'))]"
+          },
+          "hardwareProfile": {
+            "vmSize": "[parameters('masterSize')]"
+          },
+          "osProfile": {
+            "computerName": "[concat('master-vm', copyIndex())]",
+            "adminUsername": "[parameters('adminUsername')]",
+            "linuxConfiguration": {
+                "disablePasswordAuthentication": "true",
+                "ssh": {
+                    "publicKeys": [
+                        {
+                            "path": "[variables('keyPath')]",
+                            "keyData": "[parameters('adminKey')]"
+                        }
+                    ]
+                } 
+            }
+          },
+          "storageProfile": {
+            "imageReference": "[variables('imageReference')]",
+            "osDisk": {
+              "name": "osdisk",
+              "vhd": {
+                "uri": "[concat('http://',variables('masterStorageAccountName'),'.blob.core.windows.net/vhds/', variables('masterName'), copyindex(), '-osdisk.vhd')]"
```

Comment:
Because masters should not be part of part of a managed availability set. There are three fixed masters, and the master election between them is a high risk operation. The clearer the separation of concerns the better, so a more reliable pattern is for them to run off their own disks.